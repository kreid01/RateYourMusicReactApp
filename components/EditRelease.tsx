import { Formik } from "formik";
import { View, Text, Input, Button } from "native-base";
import React, { useState } from "react";
import { useUpdateReleaseMutation } from "../generated/graphql";
import { ArtistName } from "./ArtistName";
import { CheckIcon, CloseIcon } from "native-base";

interface Props {
  release: Release;
  handleEdit: () => void;
}

export type Release = {
  id: number;
  artistId: number;
  title: string;
  type: string;
  recorded: string;
  ratingCount: number;
  rating: number;
  cover: string;
  language: string;
  genres: [string];
  tracks: [string];
};

export const EditRelease: React.FC<Props> = ({ release, handleEdit }) => {
  const [, updateRelease] = useUpdateReleaseMutation();

  return (
    <Formik
      initialValues={release}
      onSubmit={(values) =>
        updateRelease({
          id: values.id,
          title: values.title,
          type: values.type,
          recorded: values.recorded,
          ratingCount: values.ratingCount,
          rating: values.rating,
          cover: values.cover,
          language: values.language,
          genres: values.genres,
          tracks: values.tracks,
        })
      }
    >
      {({ handleChange, handleBlur, handleSubmit, values }) => (
        <View>
          <View className="mt-5 ml-5 w-[90%]">
            <Input onChangeText={handleChange("title")} className=" text-3xl">
              {release.title}
            </Input>
            <View className="flex flex-row">
              <Text className="text-gray-500">
                By{" "}
                <ArtistName
                  color="text-sky-700"
                  id={release.artistId as number}
                />
              </Text>
              <View className="flex flex-row ml-auto">
                <Button
                  className="mb-5"
                  variant="ghost"
                  onPress={() => handleSubmit()}
                >
                  <CheckIcon color="#3BB143" />
                </Button>
                <Button
                  className="mb-5"
                  variant="ghost"
                  onPress={() => handleEdit()}
                >
                  <CloseIcon color="#D30000" />
                </Button>
              </View>
            </View>
          </View>

          <View className="mx-5 mt-5 p-5 bg-gray-200 ">
            <View className="flex flex-row">
              <Text className="text-gray-500 mr-8">Type:</Text>
              <Input
                width={"64"}
                height={7}
                marginTop={-1}
                onChangeText={handleChange("type")}
              >
                {release.type}
              </Input>
            </View>
            <View>
              <Text className="text-gray-500">
                Rating:{" "}
                <Text className="font-bold text-lg">{release.rating}</Text> /
                5.0 from{" "}
                <Text className="font-bold">{release.ratingCount}</Text> ratings
              </Text>
            </View>

            <View className="flex flex-row my-2">
              <Text className="text-gray-500 mr-1">Released:</Text>
              <Input
                width={"64"}
                height={7}
                marginTop={-1}
                onChangeText={handleChange("recorded")}
              >
                {release.recorded}
              </Input>
            </View>

            <View className="flex flex-row justify-between">
              <View className="flex flex-row mt-3 w-[100%] flex-wrap">
                <Text className="text-gray-500 mr-1">Genres:</Text>
                {release.genres?.map((genre, i) => (
                  <View className="-mt-1 mb-3 mx-1">
                    <Input
                      onChangeText={handleChange(`genres[${i}]`)}
                      className="text-blue-500"
                      width={24}
                    >
                      {genre}{" "}
                    </Input>
                  </View>
                ))}
              </View>
            </View>

            <View className="flex flex-row">
              <Text className="text-gray-500">Language: </Text>
              <Input
                className="text-gray-500"
                width={"64"}
                height={7}
                marginTop={-1}
                onChangeText={handleChange("language")}
              >
                {release.language}
              </Input>
            </View>

            <View>
              <Text className="text-gray-500 font-bold mb-2 mt-5">
                Track Listing
              </Text>
              <View className="border-[1px] border-gray-500">
                {release.tracks?.map((track, i) => {
                  return (
                    <View className="flex flex-row py-1 border-b-[1px] border-gray-500">
                      <Text className="text-gray-500 font-bold mx-3">
                        {i + 1}{" "}
                      </Text>
                      <Input
                        width={"72"}
                        onChangeText={handleChange(`tracks[${i}]`)}
                      >
                        {track}
                      </Input>
                    </View>
                  );
                })}
              </View>
            </View>
          </View>
        </View>
      )}
    </Formik>
  );
};
