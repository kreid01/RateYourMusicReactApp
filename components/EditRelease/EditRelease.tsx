import { Formik } from "formik";
import { View, Text, Input, Button } from "native-base";
import React, { useState } from "react";
import { useUpdateReleaseMutation } from "../../generated/graphql";
import { ArtistName } from "../ArtistName/ArtistName";
import { CheckIcon, CloseIcon } from "native-base";
import { IRelease } from "../../conts/Types";

interface Props {
  release: IRelease;
  handleEdit: () => void;
}

export const EditRelease: React.FC<Props> = ({ release, handleEdit }) => {
  const [, updateRelease] = useUpdateReleaseMutation();

  return (
    <Formik
      initialValues={release}
      onSubmit={(values) =>
        updateRelease({
          id: values.id,
          title: values.title,
          type: values.type,
          released: values.released,
          ratingCount: values.ratingCount,
          rating: values.rating,
          cover: values.cover,
          language: values.language,
          genres: values.genres,
          tracks: values.tracks,
        }).then((result) => {
          if (result.error) {
            alert(result.error.message);
          } else {
            console.debug(result);
            handleEdit();
          }
        })
      }
    >
      {({ handleChange, handleBlur, handleSubmit, values }) => {
        return (
          <View>
            <View className="mt-5 ml-5 w-[90%]">
              <View className="mb-2">
                <Input
                  className="text-white"
                  backgroundColor="#475569"
                  borderColor="#475569"
                  focusOutlineColor="#475569"
                  onChangeText={handleChange("cover")}
                  placeholder="Cover"
                >
                  {release.cover}
                </Input>
              </View>
              <Input
                backgroundColor="#475569"
                borderColor="#475569"
                focusOutlineColor="#475569"
                onChangeText={handleChange("title")}
                className="text-white text-3xl"
              >
                {release.title}
              </Input>
              <View className="flex flex-row">
                <Text className="text-gray-500">
                  By{" "}
                  <ArtistName
                    color="text-sky-700"
                    id={release.artistId as number}
                  />
                </Text>
                <View className="flex flex-row ml-auto">
                  <Button
                    className="mb-5"
                    variant="ghost"
                    onPress={() => handleSubmit()}
                  >
                    <CheckIcon color="#3BB143" />
                  </Button>
                  <Button
                    className="mb-5"
                    variant="ghost"
                    testID="edit-button"
                    onPress={() => handleEdit()}
                  >
                    <CloseIcon color="#D30000" />
                  </Button>
                </View>
              </View>
            </View>

            <View className="mx-5 mt-5 p-5 bg-gray-700 ">
              <View className="flex flex-row">
                <Text className="text-gray-100 mr-8">Type:</Text>
                <Input
                  className="text-white"
                  backgroundColor="#475569"
                  borderColor="#475569"
                  focusOutlineColor="#475569"
                  width={"64"}
                  height={7}
                  marginTop={-1}
                  onChangeText={handleChange("type")}
                >
                  {release.type}
                </Input>
              </View>
              <View>
                <Text className="text-white">
                  Rating:{" "}
                  <Text className=" text-white font-bold text-lg">
                    {release.rating}
                  </Text>{" "}
                  / 5.0 from{" "}
                  <Text className=" text-white font-bold">
                    {release.ratingCount}
                  </Text>{" "}
                  ratings
                </Text>
              </View>

              <View className="flex flex-row my-2">
                <Text className="text-white mr-1">Released:</Text>
                <Input
                  className="text-white"
                  backgroundColor="#475569"
                  borderColor="#475569"
                  focusOutlineColor="#475569"
                  width={"64"}
                  height={7}
                  marginTop={-1}
                  onChangeText={handleChange("released")}
                >
                  {release.released}
                </Input>
              </View>

              <View className="flex flex-row justify-between">
                <View className="flex flex-row mt-3 w-[100%] flex-wrap">
                  <Text className="text-white mr-1">Genres:</Text>
                  {release.genres?.map((genre, i) => (
                    <View key={i} className="-mt-1 mb-3 mx-1">
                      <Input
                        onChangeText={handleChange(`genres[${i}]`)}
                        className="text-blue-500"
                        width={24}
                      >
                        {genre}{" "}
                      </Input>
                    </View>
                  ))}
                </View>
              </View>

              <View className="flex flex-row">
                <Text className="text-white">Language: </Text>
                <Input
                  className="text-white"
                  width={"64"}
                  height={7}
                  marginTop={-1}
                  onChangeText={handleChange("language")}
                >
                  {release.language}
                </Input>
              </View>

              <View>
                <Text className="text-white font-bold mb-2 mt-5">
                  Track Listing
                </Text>
                <View className="border-[1px] border-gray-500">
                  {release.tracks?.map((track, i) => {
                    return (
                      <View
                        key={i}
                        className="flex flex-row py-1 border-b-[1px] border-gray-500"
                      >
                        <Text className="text-gray-400 font-bold mx-3">
                          {i + 1}{" "}
                        </Text>
                        <Input
                          width={"72"}
                          className="text-white"
                          onChangeText={handleChange(`tracks[${i}]`)}
                        >
                          {track}
                        </Input>
                      </View>
                    );
                  })}
                </View>
              </View>
            </View>
          </View>
        );
      }}
    </Formik>
  );
};
