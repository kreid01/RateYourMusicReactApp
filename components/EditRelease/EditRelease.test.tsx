import React from "react";
import {
  screen,
  render,
  cleanup,
  fireEvent,
} from "@testing-library/react-native";
import "@testing-library/jest-dom";
import { EditRelease } from "./EditRelease";
import { NativeBaseProvider } from "native-base";
import renderer, { act } from "react-test-renderer";
import { expect } from "@jest/globals";

const handleEdit = jest.fn();

const mockRelease = {
  title: "test",
  artistId: 0,
  id: 0,
  type: "Album",
  released: "1999 Scotland",
  ratingCount: 1,
  rating: 5,
  cover: "testimgaeurl",
  language: "English",
  tracks: ["track1", "track2"],
  genres: ["genre1", "genre2"],
};

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

afterEach(cleanup);

it("matched snapshot", async () => {
  const tree = renderer
    .create(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <EditRelease release={mockRelease} handleEdit={handleEdit} />;
      </NativeBaseProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("hides component if cancel button is clicked", async () => {
  render(
    <NativeBaseProvider initialWindowMetrics={inset}>
      <EditRelease release={mockRelease} handleEdit={handleEdit} />;
    </NativeBaseProvider>
  );
  act(() => {
    fireEvent.press(screen.getByTestId("edit-button"));
  });
  expect(handleEdit).toBeCalled();
});
