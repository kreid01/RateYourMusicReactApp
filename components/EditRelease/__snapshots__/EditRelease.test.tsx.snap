// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={{}}
  >
    <View
      className="mt-5 ml-5 w-[90%]"
      dataSet={{}}
      style={{}}
    >
      <View
        className="mb-2"
        dataSet={{}}
        style={{}}
      >
        <View
          dataSet={{}}
          isFocused={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#475569",
              "borderColor": "#475569",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "overflow": "hidden",
            }
          }
        >
          <TextInput
            accessible={true}
            className="text-white"
            dataSet={{}}
            editable={true}
            focusOutlineColor="#475569"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Cover"
            placeholderTextColor="#a3a3a3"
            secureTextEntry={false}
            style={
              {
                "backgroundColor": "transparent",
                "color": "#171717",
                "flex": 1,
                "fontFamily": undefined,
                "fontSize": 12,
                "fontStyle": "normal",
                "fontWeight": "400",
                "height": "100%",
                "paddingBottom": 8,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 8,
                "width": "100%",
              }
            }
          >
            testimgaeurl
          </TextInput>
        </View>
      </View>
      <View
        dataSet={{}}
        isFocused={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#475569",
            "borderColor": "#475569",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "overflow": "hidden",
          }
        }
      >
        <TextInput
          accessible={true}
          className="text-white text-3xl"
          dataSet={{}}
          editable={true}
          focusOutlineColor="#475569"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          placeholderTextColor="#a3a3a3"
          secureTextEntry={false}
          style={
            {
              "backgroundColor": "transparent",
              "color": "#171717",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "height": "100%",
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
              "width": "100%",
            }
          }
        >
          test
        </TextInput>
      </View>
      <View
        className="flex flex-row"
        dataSet={{}}
        style={{}}
      >
        <Text
          className="text-gray-500"
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          By
           
          <ActivityIndicator
            accessibilityLabel="loading"
            accessible={true}
            color="#0891b2"
            size="small"
            style={
              [
                [
                  {},
                  {
                    "dataSet": {},
                  },
                  {},
                ],
                undefined,
              ]
            }
            testID="spinner"
          />
        </Text>
        <View
          className="flex flex-row ml-auto"
          dataSet={{}}
          style={{}}
        >
          <View
            accessibilityRole="button"
            accessible={true}
            className="mb-5"
            collapsable={false}
            colorScheme="primary"
            dataSet={{}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "borderRadius": 4,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 10,
              }
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
              test={true}
            >
              <View
                dataSet={{}}
                style={{}}
              >
                <RNSVGSvgView
                  accessibilityRole="image"
                  align="xMidYMid"
                  bbHeight="16"
                  bbWidth="16"
                  dataSet={{}}
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  stroke=""
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "color": "#3BB143",
                        "height": 16,
                        "width": 16,
                      },
                      {
                        "flex": 0,
                        "height": 16,
                        "width": 16,
                      },
                    ]
                  }
                  tintColor="#3BB143"
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    propList={
                      [
                        "stroke",
                      ]
                    }
                    stroke={null}
                  >
                    <RNSVGGroup>
                      <RNSVGPath
                        d="M8.53115 15.1856L3.96198 10.744L2 12.6512L8.53115 19L22 5.9072L20.038 4L8.53115 15.1856Z"
                        fill={
                          {
                            "type": 2,
                          }
                        }
                        propList={
                          [
                            "fill",
                            "stroke",
                          ]
                        }
                        stroke={null}
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessible={true}
            className="mb-5"
            collapsable={false}
            colorScheme="primary"
            dataSet={{}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "borderRadius": 4,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 10,
              }
            }
            testID="edit-button"
          >
            <View
              dataSet={{}}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
              test={true}
            >
              <View
                dataSet={{}}
                style={{}}
              >
                <RNSVGSvgView
                  accessibilityRole="image"
                  align="xMidYMid"
                  bbHeight="16"
                  bbWidth="16"
                  dataSet={{}}
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  stroke=""
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "color": "#D30000",
                        "height": 16,
                        "width": 16,
                      },
                      {
                        "flex": 0,
                        "height": 16,
                        "width": 16,
                      },
                    ]
                  }
                  tintColor="#D30000"
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    propList={
                      [
                        "stroke",
                      ]
                    }
                    stroke={null}
                  >
                    <RNSVGGroup>
                      <RNSVGPath
                        d="M12 9.77778L4.22222 2L2 4.22222L9.77778 12L2 19.7778L4.22222 22L12 14.2222L19.7778 22L22 19.7778L14.2222 12L22 4.22222L19.7778 2L12 9.77778Z"
                        fill={
                          {
                            "type": 2,
                          }
                        }
                        propList={
                          [
                            "fill",
                            "stroke",
                          ]
                        }
                        stroke={null}
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      className="mx-5 mt-5 p-5 bg-gray-700 "
      dataSet={{}}
      style={{}}
    >
      <View
        className="flex flex-row"
        dataSet={{}}
        style={{}}
      >
        <Text
          className="text-gray-100 mr-8"
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Type:
        </Text>
        <View
          dataSet={{}}
          isFocused={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#475569",
              "borderColor": "#475569",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 28,
              "marginTop": -4,
              "overflow": "hidden",
              "width": 256,
            }
          }
        >
          <TextInput
            accessible={true}
            className="text-white"
            dataSet={{}}
            editable={true}
            focusOutlineColor="#475569"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholderTextColor="#a3a3a3"
            secureTextEntry={false}
            style={
              {
                "backgroundColor": "transparent",
                "color": "#171717",
                "flex": 1,
                "fontFamily": undefined,
                "fontSize": 12,
                "fontStyle": "normal",
                "fontWeight": "400",
                "height": "100%",
                "paddingBottom": 8,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 8,
                "width": "100%",
              }
            }
          >
            Album
          </TextInput>
        </View>
      </View>
      <View
        dataSet={{}}
        style={{}}
      >
        <Text
          className="text-white"
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Rating:
           
          <Text
            className=" text-white font-bold text-lg"
            dataSet={{}}
            style={
              {
                "backgroundColor": undefined,
                "fontFamily": undefined,
                "fontSize": undefined,
                "fontStyle": undefined,
                "fontWeight": undefined,
                "textDecorationLine": undefined,
              }
            }
          >
            5
          </Text>
           
          / 5.0 from
           
          <Text
            className=" text-white font-bold"
            dataSet={{}}
            style={
              {
                "backgroundColor": undefined,
                "fontFamily": undefined,
                "fontSize": undefined,
                "fontStyle": undefined,
                "fontWeight": undefined,
                "textDecorationLine": undefined,
              }
            }
          >
            1
          </Text>
           
          ratings
        </Text>
      </View>
      <View
        className="flex flex-row my-2"
        dataSet={{}}
        style={{}}
      >
        <Text
          className="text-white mr-1"
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Released:
        </Text>
        <View
          dataSet={{}}
          isFocused={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#475569",
              "borderColor": "#475569",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 28,
              "marginTop": -4,
              "overflow": "hidden",
              "width": 256,
            }
          }
        >
          <TextInput
            accessible={true}
            className="text-white"
            dataSet={{}}
            editable={true}
            focusOutlineColor="#475569"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholderTextColor="#a3a3a3"
            secureTextEntry={false}
            style={
              {
                "backgroundColor": "transparent",
                "color": "#171717",
                "flex": 1,
                "fontFamily": undefined,
                "fontSize": 12,
                "fontStyle": "normal",
                "fontWeight": "400",
                "height": "100%",
                "paddingBottom": 8,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 8,
                "width": "100%",
              }
            }
          >
            1999 Scotland
          </TextInput>
        </View>
      </View>
      <View
        className="flex flex-row justify-between"
        dataSet={{}}
        style={{}}
      >
        <View
          className="flex flex-row mt-3 w-[100%] flex-wrap"
          dataSet={{}}
          style={{}}
        >
          <Text
            className="text-white mr-1"
            dataSet={{}}
            style={
              {
                "backgroundColor": undefined,
                "color": "#171717",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          >
            Genres:
          </Text>
          <View
            className="-mt-1 mb-3 mx-1"
            dataSet={{}}
            style={{}}
          >
            <View
              dataSet={{}}
              isFocused={false}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#d4d4d4",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "overflow": "hidden",
                  "width": 96,
                }
              }
            >
              <TextInput
                accessible={true}
                className="text-blue-500"
                dataSet={{}}
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onKeyPress={[Function]}
                placeholderTextColor="#a3a3a3"
                secureTextEntry={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  }
                }
              >
                genre1
                 
              </TextInput>
            </View>
          </View>
          <View
            className="-mt-1 mb-3 mx-1"
            dataSet={{}}
            style={{}}
          >
            <View
              dataSet={{}}
              isFocused={false}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#d4d4d4",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "overflow": "hidden",
                  "width": 96,
                }
              }
            >
              <TextInput
                accessible={true}
                className="text-blue-500"
                dataSet={{}}
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onKeyPress={[Function]}
                placeholderTextColor="#a3a3a3"
                secureTextEntry={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  }
                }
              >
                genre2
                 
              </TextInput>
            </View>
          </View>
        </View>
      </View>
      <View
        className="flex flex-row"
        dataSet={{}}
        style={{}}
      >
        <Text
          className="text-white"
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Language: 
        </Text>
        <View
          dataSet={{}}
          isFocused={false}
          style={
            {
              "alignItems": "center",
              "borderColor": "#d4d4d4",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 28,
              "marginTop": -4,
              "overflow": "hidden",
              "width": 256,
            }
          }
        >
          <TextInput
            accessible={true}
            className="text-white"
            dataSet={{}}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholderTextColor="#a3a3a3"
            secureTextEntry={false}
            style={
              {
                "backgroundColor": "transparent",
                "color": "#171717",
                "flex": 1,
                "fontFamily": undefined,
                "fontSize": 12,
                "fontStyle": "normal",
                "fontWeight": "400",
                "height": "100%",
                "paddingBottom": 8,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 8,
                "width": "100%",
              }
            }
          >
            English
          </TextInput>
        </View>
      </View>
      <View
        dataSet={{}}
        style={{}}
      >
        <Text
          className="text-white font-bold mb-2 mt-5"
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Track Listing
        </Text>
        <View
          className="border-[1px] border-gray-500"
          dataSet={{}}
          style={{}}
        >
          <View
            className="flex flex-row py-1 border-b-[1px] border-gray-500"
            dataSet={{}}
            style={{}}
          >
            <Text
              className="text-gray-400 font-bold mx-3"
              dataSet={{}}
              style={
                {
                  "backgroundColor": undefined,
                  "color": "#171717",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                }
              }
            >
              1
               
            </Text>
            <View
              dataSet={{}}
              isFocused={false}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#d4d4d4",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "overflow": "hidden",
                  "width": 288,
                }
              }
            >
              <TextInput
                accessible={true}
                className="text-white"
                dataSet={{}}
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onKeyPress={[Function]}
                placeholderTextColor="#a3a3a3"
                secureTextEntry={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  }
                }
              >
                track1
              </TextInput>
            </View>
          </View>
          <View
            className="flex flex-row py-1 border-b-[1px] border-gray-500"
            dataSet={{}}
            style={{}}
          >
            <Text
              className="text-gray-400 font-bold mx-3"
              dataSet={{}}
              style={
                {
                  "backgroundColor": undefined,
                  "color": "#171717",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                }
              }
            >
              2
               
            </Text>
            <View
              dataSet={{}}
              isFocused={false}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#d4d4d4",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "overflow": "hidden",
                  "width": 288,
                }
              }
            >
              <TextInput
                accessible={true}
                className="text-white"
                dataSet={{}}
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onKeyPress={[Function]}
                placeholderTextColor="#a3a3a3"
                secureTextEntry={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  }
                }
              >
                track2
              </TextInput>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  ;
</RNCSafeAreaProvider>
`;
